### PR TITLE
DBZ-7631 Change hashing strategy to use partition columns

### PR DESCRIPTION
**Pull request by samssh**
To ensure the proper ordering of events within the same partition, it's necessary to utilize the partition columns when determining the queue index for event insertion.

### CODE REVIEW

**Comment by samssh** [comment]
https://issues.redhat.com/browse/DBZ-7631

File: cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java [hunk 1]
```diff
@@ -665,4 +665,8 @@ private List<Object> getPartitionKeys(PartitionUpdate pu) {
         return values;
     }
 
+    private int getPartitionQueueIndex(PartitionUpdate partitionUpdate) {
```
Comment (jpechane) [inline cassandra-3/src/main/java/io/debezium/connector/cassandra/Cassandra3CommitLogReadHandlerImpl.java:668]: Can this method be extracted into a utility class in core module and re-used in all three modules?

**Comment by jpechane** [comment]
@samssh Thank you for the PR. I left a comment related to a small refactoring to re-use the code.
Still this is quite breaking change and I can imagine other users prefering current behaviour. Would it be possible to create a configuration option that would control the hashing startegy? Ideally enum with two options that would allow additonal expansion if needed.

**Comment by samssh** [comment]
You're welcome.

Regarding the method refactoring, unfortunately, it's not possible to extract the method to a utility class in the core module because for every version of Cassandra we have a different artifact that contains the `PartitionUpdate` class. If you agree, I can create an abstraction and develop a wrapper class to move more common logic to the core module, but this will likely require significant changes to the codebase.




And regarding configuring the behavior, that's fine, and I will add a configuration property with an enum as soon as possible.

**Comment by jpechane** [comment]
@samssh I see, that's unfortunate. Let's forget about the refactoring then.

**Comment by samssh** [comment]
@jpechane It should be ready.

**Comment by samssh** [comment]
Thank you. Of course! I'll get on it as soon as possible.

**Pull request merged**